iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/TreeFactory.java

```diff
@@ -93,48 +94,71 @@
 @SuppressWarnings("java:S1172")
 public class TreeFactory {
 
-  public File file(Optional<List<ArgInstruction>> globalArgs, Optional<List<DockerImage>> dockerImages, Optional<SyntaxToken> spacing, SyntaxToken eof) {
+  public File file(Optional<List<ArgInstruction>> globalArgs, Optional<List<DockerImage>> dockerImages, Optional<SyntaxToken> spacingBeforeEof, SyntaxToken eof) {
     return new FileImpl(globalArgs.or(Collections.emptyList()), dockerImages.or(Collections.emptyList()), eof);
   }
 
-  public DockerImage dockerImage(FromInstruction from, Optional<List<Instruction>> instructions) {
+  public ArgInstruction fileArg(ArgInstruction arg, Optional<SyntaxToken> spacingAfter, SyntaxToken eof) {
+    return arg;
+  }
+
+  public DockerImage dockerImage(FromInstruction from, SyntaxToken eol, Optional<List<Instruction>> instructions) {
     return new DockerImageImpl(from, instructions.or(Collections.emptyList()));
   }
 
-  public Instruction instruction(Optional<SyntaxToken> spacing, Instruction instruction) {
+  public Instruction instruction(Instruction instruction) {
+    return instruction;
+  }
+
+  public Instruction instructionLine(Instruction instruction, Optional<SyntaxToken> spacingAfter, SyntaxToken eol) {
     return instruction;
   }
 
-  public OnBuildInstruction onbuild(SyntaxToken keyword, Instruction instruction) {
+  public OnBuildInstruction onbuild(SyntaxToken keyword, SyntaxToken spacing, Instruction instruction) {
```

Comment:
You don't need this if you would keep the WHITESPACE in the DockerLexicalGrammar.